crates/ruff_wasm/Cargo.toml

```diff
@@ -1,6 +1,6 @@
 [package]
-name = "ruff_wasm"
-version = "0.0.0"
+name = "ruff-api"
```

Comment:
I would prefer to keep this named `ruff_wasm`. We may have other API crates in the future that aren't wasm specific.

Reply:
Would you still want to publish as `@astral-sh/ruff-api`? The crate name is used in the generated `package.json`, but there can be a CI step to rewrite that to use the preferred name for publishing

Reply:
Possibly. It depends on what we want to do about the different targets.

Reply:
It looks like wasm-pack creates a different wasm file depending on the target (not just the JS glue code)

Each of those wasm files is ~7.5MB. So publishing all under 1 package would mean a developer needs to download ~30MB. While that's not massive in terms of node_modules size, it's still pretty big and would be mostly unused

So it might be better to upload separate packages per target. Also simplifies the CI since it could in theory just use a matrix w/ the targets and publish as `@astral-sh/ruff-api-{target}`

As an aside, projects shipping Rust native binaries on npm (such as [swc](https://www.npmjs.com/package/@swc/core) often publish a single entry package w/ optional dependencies. This takes advantage of npm optional dependencies failing to install if the dependency package.json specifies a specific architecture requirement.

The same can't be used here since web/node is not something you can specify with npm, but just pointing it out that it seems other projects tend to go towards multiple packages. Although the binaries for swc are 50MB and there's like 10 architectures, so in that case they definitely don't want to push 450MB of unusable code to every user.

Reply:
That makes sense and also roughly matches what Biome does (one `js-api` package that has optional dependencies on `wasm-node` and `wasm-web`). I think it's okay to make it the users responsibility to install the optional dependencies (which we don't need for now). 

I'm leaning towards naming the packages `ruff-wasm-node`, and `ruff-wasm-web`. That would allow us to create a `ruff-api` package long term that wraps the glue code with a proper API. But I don't mind much either way.

Reply:
Oh neat I learned something new from looking at their package.

Npm used to not install `peerDependencies` by default. They changed it in v7 to install them by default. Looks like now there's a `peerDependenciesMeta` field which can mark as optional so it doesn't install those by default on v7+